### DISCUSSION TITLE
How to get auto generated USDZ work?

### DISCUSSION THREAD

**Discussion by richardrevesz** [google/model-viewer#2391] [Q&A] [answered]
Hi! 
I was thrilled to read Emett's update the new release (1.7.0.) and read that auto generated USDZ it is meant to work with Scene Graph swap textures.
Thing is, tapping on the AR button does not do anything on my iPad (A1701), it refreshes something (not the page, maybe the model viewer?) and that's it.
I have iPadOS 14.5.1 and used the latest Chrome (updated today from appstore).
If I don't swap textures simply just use a glb and remove ios-src, the result is the same.
What I am missing?
Thank you in advance!
Richard

**Comment by colordrop** [comment]
I have a similar problem. I can't start AR on some iOS devices, tested on multiple iPad's and iPhones - AR works only on some of them. The problem occurs only on model-viewer with swappable textures. Tapping the AR button reloads the model and after that button disappears. 
On my iPadOS 14.5.1 works fine only on Safari - on Chrome models with texture swap, don't work in AR but there are few devices with the problem described above on Safari.
You can check my tests here:
https://colordrop.pl/ar-iker/

**Reply by elalish** [reply]
Interesting, yes this looks like a Chrome iOS bug; it can send a normal URL to Quick Look (ios-src), but doesn't seem to be able to handle a blob URL. I'll file a bug on Chrome.

If you have a specific problem with Safari, please open an issue and complete the whole template, as that will be unrelated to this Chrome issue.

**Reply by richardrevesz** [reply]
Yes, tried in Safari and working fine, thank you!

**Comment by richardrevesz** [comment, marked as answer]
As @elalish found out, it is an Chrome iOS bug, works fine on Safary.
- Commenting this here as I can not mark his comment as answer -

**Reply by elalish** [reply]
Here's the Chrome iOS bug: https://bugs.chromium.org/p/chromium/issues/detail?id=1211526
